src/pe/authenticode.rs

```diff
@@ -74,28 +95,164 @@ impl<'s> Iterator for ExcludedSectionsIter<'s> {
 
         if let Some(sections) = self.pe.authenticode_excluded_sections.as_ref() {
             loop {
-                match self.state {
+                match mem::replace(&mut self.state, IterState::Pending) {
```

Comment:
The only reason I need to mem::replace (and make data moves) is because I'm consuming the sections progressively and it's a member of `IterState`.

I could avoid that be making `sections: VecDeque<_>` a member of `ExcludedSectionsIter` instead and just mutate it.